json-endpoints/class.wpcom-json-api-update-site-homepage-endpoint.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+class WPCOM_JSON_API_Update_Site_Homepage_Endpoint extends WPCOM_JSON_API_Endpoint {
+
+	function callback( $path = '', $site_id = 0 ) {
+		$blog_id = $this->api->switch_to_blog_and_validate_user( $this->api->get_blog_id( $site_id ) );
+		if ( is_wp_error( $blog_id ) ) {
+			return $blog_id;
+		}
+
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			return new WP_Error( 'unauthorized', 'User is not authorized to access homepage settings', 403 );
```

Comment:
Agree with @sirbrillig, much of WPCOMs REST error messaging is not translated like this, because these cases are more for developers to see than for end users.